presto-spi/src/main/java/com/facebook/presto/spi/session/SessionPropertyConfigurationManager.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.session;
+
+import java.util.List;
+
+public interface SessionPropertyConfigurationManager
+{
+    List<SessionPropertyRule> getRulesInPriorityOrderLowToHigh();
```

Comment:
@electrum @rongrong how do you feel about renaming this to `getRules` and let the user of this interface sort it if necessary? 

I feel like this name sounds strange. If we really need the implementors to return a sorted list, then we can still name this method `getRules` and specify the contract of this interface in javadoc and say that implementors must return a sorted list etc.

Reply:
That's what I did originally, but then it means that either QuerySessionSupplier has to cache a copy of the sorted one, or it has to sort them every time. I guess I can put the cache in the session supplier.

Reply:
Let's name the method `getOrderedRules()` and explain the ordering in the Javadoc (which we should have for every SPI method). I'm not sure what "low to high" means. Do we evaluate the highest priority rules first (and thus the list is evaluated in reverse order)?

So, even with a very long name, we still need Javadoc, so let's go with the shorter (and more conventional) name.